backend/clap/src/main/java/zpi/ppea/clap/controller/SubmissionController.java

```diff
@@ -19,21 +22,43 @@
 @ControllerAdvice
 public class SubmissionController {
 
-    private final SubmissionService submissionService;
+    private final DataStoreService dataStoreService;
+    private final FirebaseAgent agent;
 
     @GetMapping
-    public ResponseEntity<List<SubmissionDto>> getSubmissions() {
-        return ResponseEntity.ok(submissionService.getSubmissions());
+    public ResponseEntity<List<SubmissionDto>> getSubmissions(
+        @RequestHeader(name = "Authorization") String bearerToken
+    ) {
+
+        var auth = agent.authenticate(bearerToken);
+        return ResponseEntity.ok()
+            .header("refresh", auth.getRefresh())
+            .body(dataStoreService.getSubmissions(auth.getClaims().getAssessorId()));
     }
```

Comment:
What in case if `getSubmissions` results in an error? Will the `refresh` header be added correctly?

Reply:
This should be a common occurrence, but i added a path for this. Now Error handlers take into consideration `refresh` header.